README.md

```diff
@@ -1,6 +1,36 @@
 # consul-deployment-agent
 
-#### Command line options
+A cross platform service that synchronises the set of applications installed on a machine with a manifest stored in a Consul key value store.
+
+## Package Format
+
+consul-deployment-agent installs applications packaged in the AWS [CodeDeploy revision format](http://docs.aws.amazon.com/codedeploy/latest/userguide/reference-appspec-file.html). This is a `.zip` file containing your code and an `appspec.yml` file that describes how to install the application.
+
+### Hook Scripts
+
+consul-deployment-agent supports the following subset of the [deployment lifecycle hooks](http://docs.aws.amazon.com/codedeploy/latest/userguide/reference-appspec-file-structure-hooks.html):
+
+- BeforeInstall
+- AfterInstall
+- StartApplication
+- ValidateService
+
+#### Hook Script Execution Environment
+
+consul-deployment-agent sets the following variables in the hook script execution environment
+
+| Variable | Value |
+|---|---|
+| APPLICATION_ID | ? |,
+| DEPLOYMENT_BASE_DIR | ? |
```

Comment:
This is the archive directory of the deployment. Depending on the target platform:

Linux: /opt/consul-deployment-agent/deployments
Windows: C:\TLDeploy

On start hook, it's the archive dir in that directory for this deployment. 
On stop hook, it's the last archive dir for this deployment.